apps/openmw/mwrender/npcanimation.cpp

```diff
@@ -511,6 +513,54 @@ void NpcAnimation::updateNpcBase()
     mWeaponAnimationTime->updateStartTime();
 }
 
+std::string NpcAnimation::getShieldMesh(MWWorld::ConstPtr shield) const
+{
+    std::string mesh = shield.getClass().getModel(shield);
+    const ESM::Armor *armor = shield.get<ESM::Armor>()->mBase;
+    std::vector<ESM::PartReference> bodyparts = armor->mParts.mParts;
+    if (!bodyparts.empty())
+    {
+        // For NPCs try to get shield model from bodyparts first, with ground model as fallback
+        for (auto & part : bodyparts)
+        {
+            if (part.mPart != ESM::PRT_Shield)
+                continue;
+
+            std::string bodypartName;
+            if (!mNpc->isMale() && !part.mFemale.empty())
+                bodypartName = part.mFemale;
+            else if (!part.mMale.empty())
+                bodypartName = part.mMale;
+
+            if (!bodypartName.empty())
+            {
+                const MWWorld::ESMStore &store = MWBase::Environment::get().getWorld()->getStore();
+                const MWWorld::Store<ESM::BodyPart> &partStore = store.get<ESM::BodyPart>();
```

Comment:
Not sure but may it be worth it to move the code out of the for each loop or is performance gain too negligible?